File: parts/p002/src/main/java/org/rapidpm/event/mutationtesting/p002/Service.java
package org.rapidpm.event.mutationtesting.p002;

/**
 * Create the best test coverage you can.
 * How many tests you will need?
 * What are the "good" tests?
 */
public class Service {

  public int add(int a , int b) {
    if (a < 2) {
      return (a + b) * - 1;
    } else {
      return a + b;
    }
  }

  public int subAbs(int a, int b) {
      int aAbs = Math.abs(a);
      int bAbs = Math.abs(b);
      return  a < b ? aAbs - bAbs : bAbs - aAbs;
  }
}

File: parts/p002/src/main/kotlin/org/rapidpm/event/mutationtesting/p002/Service.kt
package org.rapidpm.event.mutationtesting.p002

/**
 * Create the best test coverage you can.
 * How many tests you will need?
 * What are the "good" tests?
 */
class Service {

    fun add(a: Int, b: Int): Int {
        return when {
            a < 2 -> (a - b) * -1
            b > 0 -> b - a
            else -> a + b
        }
    }
}
